src/common.rs

```diff
@@ -87,19 +96,93 @@ where
     }
 }
 
+/// Struct used to detect whether or not the `.done()` method was called.
+#[derive(Debug)]
+struct DoneCallDetector {
+    called: bool,
+}
+
+impl DoneCallDetector {
+    fn new() -> Self {
+        Self { called: false }
+    }
+
+    /// Mark the `.done()` method as called.
+    ///
+    /// Note: When calling this method twice, an assertion failure will be
+    /// triggered.
+    fn mark_as_called(&mut self) {
+        assert!(!self.called, "The `.done()` method was called twice!");
+        self.called = true;
+    }
+}
+
+impl Drop for DoneCallDetector {
+    fn drop(&mut self) {
+        // Ensure that the `.done()` method was called on the mock before
+        // dropping.
+        if !self.called && !thread::panicking() {
+            let msg = "WARNING: A mock (from embedded-hal-mock) was dropped \
+                       without calling the `.done()` method. \
+                       See https://github.com/dbrgn/embedded-hal-mock/issues/34 \
+                       for more details.";
+
+            // Note: We cannot use the print macros here, since they get
+            // captured by the Cargo test runner. Instead, write to stderr
+            // directly.
+            use std::io::Write;
+            let mut stderr = std::io::stderr();
+            stderr.write_all(b"\x1b[31m").ok();
+            stderr.write_all(msg.as_bytes()).ok();
+            stderr.write_all(b"\x1b[m\n").ok();
+            stderr.flush().ok();
+
+            // Panic. This probably results in an abort:
+            // https://doc.rust-lang.org/std/ops/trait.Drop.html#panics
+            panic!("{}", msg);
```

Comment:
Why should this result in an abort? Only if we *not* check for `!thread::panicking`, right?

Reply:
Yeah, I think you're right.

Reply:
Note updated in https://github.com/dbrgn/embedded-hal-mock/pull/79